lib/exhal/navigation.ex

```diff
@@ -12,9 +12,8 @@ defmodule ExHal.Navigation do
   `{:error, %ExHal.Error{...}}` if not
   """
   def follow_link(a_doc, name, opts \\ %{tmpl_vars: %{}, strict: false, headers: []}) do
-    opts = Map.new(opts)
+    {tmpl_vars, opts} = tmpl_vars_and_opts_map(opts)
     pick_volunteer? = !Map.get(opts, :strict, false)
```

Comment:
We should put the `strict` extraction in `tmpl_vars_and_opts` too. Maybe rename it to `interpret_nav_opts` or something?

Reply:
done